### PR TITLE
Initial Commit - Ansible role for Software Bill of Material generation for any project

### PR DESCRIPTION
**Pull request by Pranshu021**
Lifecycle features added :
- oah_install.yml
- oah_validate.yml
- oah_update.yml
- oah_remove.yml

@asa1997 Please review

### CODE REVIEW

File: tasks/oah-install.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+
+- name: Checking the latest version
+  shell: 'curl -s https://api.github.com/repos/opensbom-generator/spdx-sbom-generator/releases/latest | grep "browser_download_url.*linux-amd64.tar.gz" | cut -d : -f 2,3 | tr -d \" | head -1 | xargs'
+  register: latest_version
+  warn: false
```
Comment (asa1997) [inline tasks/oah-install.yml]: Conflicting action statements: shell, warn

File: tasks/oah-update.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Checking latest version
+  shell: 'curl -s https://api.github.com/repos/opensbom-generator/spdx-sbom-generator/releases/latest | grep "browser_download_url.*linux-amd64.tar.gz" | cut -d : -f 2,3 | tr -d \" | head -1 | xargs'
+  register: update_link
+  warn: false
```
Comment (asa1997) [inline tasks/oah-update.yml]: Conflicting action statements: shell and warn

File: tasks/oah-install.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+
+- name: Checking the latest version
+  shell: 'curl -s https://api.github.com/repos/opensbom-generator/spdx-sbom-generator/releases/latest | grep "browser_download_url.*linux-amd64.tar.gz" | cut -d : -f 2,3 | tr -d \" | head -1 | xargs'
+  register: latest_version
+  warn: false
+
+- name: Download and Extract spdx-generator tool
```
Comment (asa1997) [inline tasks/oah-install.yml:13]: Require sudo

File: tasks/oah-validate.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: Validate Installation
+  stat:
+    path: "/opt/spdx-sbom-generator"
+  register: spdx_executable 
+
+- name: Report if file doesn't exist
```
Comment (asa1997) [inline tasks/oah-validate.yml:8]: Please include a message for when the tool is found

File: tasks/oah-update.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Checking latest version
+  shell: 'curl -s https://api.github.com/repos/opensbom-generator/spdx-sbom-generator/releases/latest | grep "browser_download_url.*linux-amd64.tar.gz" | cut -d : -f 2,3 | tr -d \" | head -1 | xargs'
+  register: update_link
+  warn: false
+  
+- name: Checking if latest version is installed
+  debug:
+    msg: "Latest version already installed"
+  when: "'{{ update_link.stdout }}' == '{{ release_link }}/{{ latest_version }}'"
```
Comment (asa1997) [inline tasks/oah-update.yml]: Syntax incorrect.

image

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+# tasks file for spdx-generator
+- include: "tasks/oah-install.yml"
+  when: oah_command == "install"
+
+- include: "tasks/oah-remove.yml"
+  when: oah_command == "uninstall"
```
Comment (asa1997) [inline tasks/main.yml]: Wrong command. Should be `remove`.

File: tasks/oah-remove.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+
+- name: Removing spdx-generator
```
Comment (asa1997) [inline tasks/oah-remove.yml:3]: Require sudo

**Review by asa1997 — CHANGES_REQUESTED** [review]
@Pranshu021 Please address the comments.

File: tasks/oah-install.yml (hunk 1 above)
Comment (Pranshu021) [reply tasks/oah-install.yml]: Removed the conflicited warn statement

File: tasks/oah-install.yml (hunk 3 above)
Comment (Pranshu021) [reply tasks/oah-install.yml:13]: Added the required statement

Comment (Pranshu021) [reply tasks/oah-install.yml:13]: Added the required sudo statement for curs -s also

File: tasks/oah-validate.yml (hunk 4 above)
Comment (Pranshu021) [reply tasks/oah-validate.yml:8]: Added a message for when tool is found

Comment (Pranshu021) [reply tasks/oah-validate.yml:8]: Replaced a debug message with fail statement when tool is not found

File: tasks/main.yml (hunk 6 above)
Comment (Pranshu021) [reply tasks/main.yml]: Replaced uninstall with remove command

File: tasks/oah-remove.yml (hunk 7 above)
Comment (Pranshu021) [reply tasks/oah-remove.yml:3]: Added the required permission statement

**Comment by Pranshu021** [comment]
@asa1997 Changes made, Please check

File: tasks/oah-update.yml (hunk 5 above)
Comment (Pranshu021) [reply tasks/oah-update.yml]: Changed the when condition which was creating this Syntax error. Now working fine.

**Comment by Pranshu021** [comment]
@asa1997 Made the changes. Please check

File: tasks/oah-update.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+- name: Checking installation
+  include: "tasks/oah-validate.yml"
+  
+- name: Checking if latest version is installed
+  debug:
+    msg: "Latest version already installed"
+  when: '"{{update_link.stdout}}" == "{{release_link}}/{{latest}}"'
+
+- name: Updating to latest version
```
Comment (asa1997) [inline tasks/oah-update.yml:14]: Error.
image

**Review by asa1997 — CHANGES_REQUESTED** [review]
@Pranshu021 Please address the comments.

File: tasks/oah-update.yml (hunk 8 above)
Comment (Pranshu021) [reply tasks/oah-update.yml:14]: @asa1997 Please install curl in your system. Error shows that it couldn't fetch the source from the link. 
Although I am adding the functionality to check and install curl in install.yml.

Comment (Pranshu021) [reply tasks/oah-update.yml:14]: Added the validation for curl and install it if not present in install.yml (commit : c153afe6a0c10ecde90ce966bb25f26b9d614eda)

**Comment by Pranshu021** [comment]
@asa1997 Made the changes, Created README.md file and added oah-reset.yml
Please check

File: tasks/oah-update.yml (hunk 8 above)
Comment (asa1997) [reply tasks/oah-update.yml:14]: Error has been resolved.

**Pull request closed without merging**